src/object/srv_obj.c

```diff
@@ -1537,6 +1544,12 @@ ds_obj_punch_handler(crt_rpc_t *rpc)
 			D_GOTO(out, rc);
 	}
 
+	/* FIXME: until distributed transaction. */
+	if (opi->opi_epoch == DAOS_EPOCH_MAX) {
+		opi->opi_epoch = daos_ts2epoch();
+		D_DEBUG(DB_IO, "overwrite epoch "DF_U64"\n", opi->opi_epoch);
+	}
+
```

Comment:
need similar handling for ds_obj_enum_handler() for kinds of obj_list/enumerate.